Updated/Added PE block definition files for Structure Block

StealthyExpertX opened the pull request:
Added Structure Block Texture and ID to MCEdit


Added block id for PE structure block pocketMaterials.StructureBlock = pocketMaterials[252, 0] on line 1168 in materials.py at path \MCEdit-Unified\pymclevel\materials.py

Added and updated block texture for PE structure block 16x16pxs to pocket-terrain.png at index[9, 19] of the image file at path \MCEdit-Unified\pocket-terrain.png

Added definitions for (name, id, mapcolor, texture) for PE structure block to pocket.json at path  
\MCEdit-Unified\pymclevel\pocket.json

Added definitions for (name, id, mapcolor, texture) for PE structure block to blocks.json at path
\MCEdit-Unified\mcver\PE\blocks.json

LaChal commented on pymclevel/materials.py:
```diff
@@ -944,230 +944,230 @@ def build_indev_materials():
 
 # --- Pocket static block defs ---
 def build_pocket_materials():
-    log.info("Building Pocket materials.")
```
Please, pay attention to the indentation character.
We use four spaces for indentation, not tabulation character.
This is important, since Python programs will behave strangely an  may crash if the indentation levels are not strictly respected.

StealthyExpertX replied:
okay i'll edit the code and repush.

StealthyExpertX commented:
I have fixed the missing comma error in the json file as well.

StealthyExpertX commented:
fixed the missing comma and retested the build. textures, json files, ect. all working as intended 
added the PE structure block definition files can you give me an example of the quality that you want
because I'm still very new to github and helping with mcedit's development.
?

LaChal commented:
You could let the PR opened 😉 

Firstly, was it intentional to remove the line 1111 in materials.py (`pocketMaterials.Tripwire = pocketMaterials[132, 0]`)? If yes can you tell why?

Regarding quality, we aim on functional stuff firstly.
So:
* Don't commit broken code.
* Be *very careful* when touching code which is used for different game versions. By fixing something for PC 1.12, it is possible to break something for PC 1.10.2 and/or PE.
* Test your code as much as possible.
* Keep your code as much as possible platform agnostic.
  For example: instead of opening a file with `open("foo\bar\baz")`, use something like `open(os.path.join("foo", "bar", "baz")`. Python will automagically use the right path for the OS.
  We have some internal functions which can simplfy few things, like opening a dialog for file selection.
* We're not linter integrists, but we tend to adopt some good practices in code formatting to keep it readable.
* Using an IDE like PyPE, PyCharm or Eclipse/PyDev can help :smile:

StealthyExpertX commented:
I did check for any missing materials I'm not really sure why its missing I
didn't remove the line 1111 so it must be unintentional even though which
is funny is that it shows in mcedit when the line doesn't exist?
ill re/add the missing line and repush the code from the desktop github
client.

[image: Inline image 1]

StealthyExpertX commented:
Ill compare all code with originals before pushing it was unintentional A side effect from fixing the incorrect indentations from before.  and I did not make any real changes to the code except for line 1167 so if that's all that was wrong it's now corrected and will no longer have any issues,
